Networking/NetworkingTests/Remote/BlazeRemoteTests.swift

```diff
@@ -310,4 +310,61 @@ final class BlazeRemoteTests: XCTestCase {
             XCTAssertEqual(error as? NetworkError, expectedError)
         }
     }
+
+    // MARK: - Fetch forecasted impressions
+
+    func test_fetchForecastedImpressions_returns_parsed_impressions() async throws {
+        // Given
+        let remote = BlazeRemote(network: network)
+        let suffix = "sites/\(sampleSiteID)/wordads/dsp/api/v1.1/forecast"
+        network.simulateResponse(requestUrlSuffix: suffix, filename: "blaze-impressions")
+
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd"
+
+        // When
+        let result = try await remote.fetchForecastedImpressions(
+            for: sampleSiteID,
+            with: BlazeForecastedImpressionsInput(startDate: dateFormatter.date(from: "2023-12-5")!,
```

Comment:
Suggestion: You can make `BlazeForecastedImpressionsInput` conform to `GeneratedFakeable` to fake a value instead of creating the whole struct for testing.

Reply:
Good idea, added in https://github.com/woocommerce/woocommerce-ios/pull/11573/commits/8b64e94d1116b62443096675a16f27b349cbce32